### PR TITLE
Fix: cue template health check skipStandard workload and fix test diverge

### PR DESCRIPTION
**Pull request by wangyikewxgm**
Signed-off-by: wangyike <wangyike_wyk@163.com>

cue template health check skipStandard workload and fix rollout e2e-test diverge

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/application/apply.go [hunk 1]
```diff
@@ -223,7 +223,7 @@ func (h *AppHandler) collectHealthStatus(wl *appfile.Workload, appRev *v1beta1.A
 			isHealth = true
 		}
 		status.Message = configuration.Status.Apply.Message
-	} else {
+	} else if !skipStandardWorkload {
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/apply.go:226]: please add an annotation to explain here

File: test/e2e-test/rollout_trait_test.go [hunk 2]
```diff
@@ -53,26 +52,6 @@ var _ = Describe("rollout related e2e-test,rollout trait test", func() {
...
-			},
-			time.Second*3, time.Millisecond*300).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
-		var td v1beta1.TraitDefinition
-		Expect(yaml.Unmarshal([]byte(rolloutTestTd), &td)).Should(BeNil())
-		td.Namespace = namespaceName
-		Eventually(func() error {
-			return k8sClient.Create(ctx, &td)
-		},
-			time.Second*3, time.Millisecond*300).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
-	}
```
Comment (wonderflow) [inline test/e2e-test/rollout_trait_test.go:74]: why deleted？

File: pkg/controller/core.oam.dev/v1alpha2/application/apply.go [hunk 3]
```diff
@@ -192,7 +192,7 @@ func (h *AppHandler) ProduceArtifacts(ctx context.Context, comps []*types.Compon
 	return h.createResourcesConfigMap(ctx, h.currentAppRev, comps, policies)
 }
 
-func (h *AppHandler) collectHealthStatus(wl *appfile.Workload, appRev *v1beta1.ApplicationRevision, overrideNamespace string) (*common.ApplicationComponentStatus, bool, error) {
+func (h *AppHandler) collectHealthStatus(wl *appfile.Workload, appRev *v1beta1.ApplicationRevision, overrideNamespace string, skipStandardWorkload bool) (*common.ApplicationComponentStatus, bool, error) {
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/apply.go:195]: skipStandardWorkload 能不能作为参数放到 appfile.Workload 里面？ 感觉整体都很丑，包括下面的 else if , 应该做成  wl.Evalhealth 的时候判断 是不是被skip的就返回

File: test/e2e-test/rollout_trait_test.go (hunk 2 above)
Comment (wangyikewxgm) [reply test/e2e-test/rollout_trait_test.go:74]: Before we write some specify component/trait definitions for these e2e-test. This due to some changes of  x-def won't be used by our e2e-tests.  Delete these means using all official x-def run our test. Guarantee  this is truly e2e-test.

File: pkg/controller/core.oam.dev/v1alpha2/application/apply.go (hunk 3 above)
Comment (wangyikewxgm) [reply pkg/controller/core.oam.dev/v1alpha2/application/apply.go:195]: 这个函数里包含了检查 workload和trait，skipStandardWorkload 这个参数只是表示要跳过 workload的测试，其他的trait还是要继续check的

Comment (wonderflow) [reply pkg/controller/core.oam.dev/v1alpha2/application/apply.go:195]: 放到 appfile.workload 里面也是只针对 workload 本身的检查跳过

File: test/e2e-test/rollout_trait_test.go (hunk 2 above)
Comment (wonderflow) [reply test/e2e-test/rollout_trait_test.go:74]: good.

File: pkg/controller/core.oam.dev/v1alpha2/application/apply.go (hunk 1 above)
Comment (wangyikewxgm) [reply pkg/controller/core.oam.dev/v1alpha2/application/apply.go:226]: added

File: pkg/controller/core.oam.dev/v1alpha2/application/apply.go (hunk 3 above)
Comment (wangyikewxgm) [reply pkg/controller/core.oam.dev/v1alpha2/application/apply.go]: fixed

**Review by wonderflow — APPROVED** [review]

**Pull request merged**